File: src/main/java/org/sonar/plugins/xquery/parser/node/Import.java
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.parser.node;

import java.util.ArrayList;
import java.util.List;

public class Import extends Declaration {

    private List<String> atHints;

    public Import(String name, String namespace) {
        super(name, namespace);
        atHints = new ArrayList<String>();
    }

    public void addAtHint(String hint) {
        atHints.add(hint);
    }
    
    public String getAtHint() {
        return getAtHint(0);
    }
    
    public String getAtHint(int index) {
        if (atHints.size() > index) {
            return atHints.get(index);
        }
        return null;
    }

    public List<String> getAtHints() {
        return atHints;
    }

    public void setAtHints(List<String> atHints) {
        this.atHints = atHints;
    }
}

File: src/main/java/org/sonar/plugins/xquery/parser/node/Import.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.parser.node

import java.util.ArrayList

class Import(name: String, namespace: String) : Declaration(name, namespace) {

    private var atHints: MutableList<String>? = null

    val atHint: String?
        get() = getAtHint(0)

    init {
        atHints = ArrayList()
    }

    fun addAtHint(hint: String) {
        atHints!!.add(hint)
    }

    fun getAtHint(index: Int): String? {
        return if (atHints!!.size > index) {
            atHints!![index]
        } else null
    }

    fun getAtHints(): List<String>? {
        return atHints
    }

    fun setAtHints(atHints: MutableList<String>) {
        this.atHints = atHints
    }
}
